src/pe/header.rs

```diff
@@ -149,6 +161,41 @@ impl CoffHeader {
         let length = bytes.pread_with::<u32>(offset, scroll::LE)? as usize;
         Ok(strtab::Strtab::parse(bytes, offset, length, 0)?)
     }
+
+    /// Validate the `CoffHeader`.
+    ///
+    /// `is_image` being `true` indicates the `CoffHeader` is in a `PE`,
+    /// being `false` indicates the `CoffHeader` is in a `Coff`.
+    pub fn validate(&self, is_image: bool) -> error::Result<()> {
```

Comment:
this suffers from boolean blindness, and is particularly difficult because there's nothing really boolean about how it affects the semantics of validation here.  Not sure what the best solution is, but because it's a (new) pub method, would like to think about this better.